creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -347,129 +424,272 @@ class SnapbackSM {
+      let randomHealthyNode = _.sample(healthyNodes)
+
+      // If node is already present in new replica set or is part of the exiting replica set, keep finding a unique healthy node
+      if (newReplicaNodesSet.has(randomHealthyNode) || healthyReplicaSet.has(randomHealthyNode)) continue
 
       // Check to make sure that the newly selected secondary does not have existing user state
       try {
-        const clockValue = await CreatorNode.getClockValue(newSecondary, wallet)
-        if (clockValue === -1) currentNodeHasExistingUserState = false
+        const clockValue = await CreatorNode.getClockValue(randomHealthyNode, wallet)
+        if (clockValue === -1) { newReplicaNodesSet.add(randomHealthyNode) }
       } catch (e) {
         // Something went wrong in checking clock value. Reselect another secondary.
-        this.logError(e.message)
+        this.logError(`[selectRandomReplicaSetNode] ${e.message}`)
       }
     }
 
-    if (currentNodeHasExistingUserState) throw new Error(`Unable to select new secondary given current healthy Content Nodes=[${healthyNodes}]`)
+    if (newReplicaNodesSet.size < numberOfUnhealthyReplicas) {
+      throw new Error('[selectRandomReplicaSetNode] Not enough healthy nodes found to issue new replica set')
+    }
 
-    const currentHealthySecondary = unhealthyReplicasSet.has(secondary1) ? secondary2 : secondary1
+    return Array.from(newReplicaNodesSet)
+  }
 
-    // NOTE: For v0 implementation, `newSecondary2` will be the newly selected secondary, whilst the other nodes will be the
-    // original replica set primary and secondary1
-    return {
-      newPrimary: primary,
-      newSecondary1: currentHealthySecondary,
-      newSecondary2: newSecondary
+  /**
+   * In the case that the primary is in `unhealthyReplicasSet`, attempt last ditch effort to hit the health
+   * check verbose route to see if it responds with a 200. If so, mark the primary as healthy. We do this as to not eagerly
+   * move user data off of a primary as this is an expensive operation we do not want to frequently perform.
+   *
+   * Additionally, check the secondaries clock values. If the values are null, add the secondaries to the unhealthy set.
+   *
+   * @param {Object} param
+   * @param {Set<string>} unhealthyReplicasSet set of current replica set endpoints that are unhealthy
+   * @param {string} primary endpoint of the current primary node on replica set
+   * @param {string} secondary1 endpoint of the current first secondary node on replica set
+   * @param {string} secondary2 endpoint of the current second secondary node on replica set
+   * @param {string} wallet wallet address of user id
+   * @param {Object} secondariesToClockMap mapping of { secondary endpoint : clock value }. The return response from `fetchClockValues()`
+   * @returns {Set<string>} updated set of unhealthy replica set endpoints
+   */
```

Comment:
this function is no longer used and can be removed, maybe worth checking other ones as well

Reply:
whoops, thought I removed this. removing